src/System.Management.Automation/engine/Modules/SwitchProcessCommand.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Runtime.InteropServices;
+
+using Dbg = System.Management.Automation.Diagnostics;
+
+#if UNIX
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// Implements a cmdlet that allows use of execv API.
+    /// </summary>
+    [Experimental(ExperimentalFeature.PSExecFeatureName, ExperimentAction.Show)]
+    [Cmdlet(VerbsCommon.Switch, "Process", HelpUri = "https://go.microsoft.com/fwlink/?linkid=2181448")]
+    public sealed class SwitchProcessCommand : PSCmdlet
+    {
+        /// <summary>
+        /// Get or set the command and arguments to replace the current pwsh process.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = false, ValueFromPipeline = true, ValueFromRemainingArguments = true)]
+        public string[] WithCommand { get; set; } = Array.Empty<string>();
+
+        /// <summary>
+        /// For each record, execute it, and push the results into the success stream.
+        /// </summary>
+        protected override void ProcessRecord()
```

Comment:
It's a little tricky if you want to support pipeline input, and the way it currently works can only support executing a piped in command without any parameters.

Imagine a user wants to run `uname -a` with pipeline input, then what should be piped in to `Switch-Process`? `uname -a` as a string won't work, because `switch-process` will treat that string as the command name. Passing in 2 strings `uname` and `-a` won't work too, because when the `ProcessRecord` runs on the first pipeline object `uname`, it will start that process, without the desired parameter.

So, I think if you want to support pipeline objects, you may need to collect all pipeline objects in `ProcessRecord`, and the code that calls `exec` should be put in `EndProcessing` instead.

Reply:
I don't think we would want to support pipeline input for this case.  The primary use case here is `exec` compatibility as some tools on Unix systems expect that to exist.

Reply:
I think that's fine. Then please update the PR to remove the code related to pipeline input processing, including:
1. update parameter attribute
2. change `ProcessRecord` to `EndProcessing`.